### PR TITLE
fix: forbidUnknownValues should default true when validatorOptions undefined

### LINKED ISSUE #1906: fix: forbidUnknownValues doesn't default to true when validatorOptions are undefined
### Description
0.14.0 is supposed to make `forbidUnknownValues` default to `true`. However, it only defaults to `true` when a `validatorOptions` object is passed.

**Minimal code-snippet showcasing the problem**
```ts
await validateOrReject({}); // resolves
await validateOrReject({}, {}); // rejects
```

### Expected behavior
`forbidUnknownValues` should default to `true` unless explicitly specified as `false`.

### Actual behavior
`forbidUnknownValues` continues to default to `false` when no `validatorOptions` object is specified.

### PR DESCRIPTION
**Pull request by cduff**
## Description

`forbidUnknownValues` is supposed to default to `true`. This wasn't the case when `validatorOptions` was `undefined`. This PR corrects that and adds a test for it. It also corrects some related code doc.

## Checklist

- [x] the pull request title describes what this PR does (not a vague title like `Update index.md`)
- [x] the pull request targets the *default* branch of the repository (`develop`)
- [x] the code follows the established code style of the repository
  - `npm run prettier:check` passes
  - `npm run lint:check` passes
- [x] tests are added for the changes I made (if any source code was modified)
- [x] documentation added or updated
- [x] I have run the project locally and verified that there are no errors

### Fixes

fixes #1906

### CODE REVIEW

File: src/validation/ValidatorOptions.ts [hunk 1]
```diff
@@ -72,10 +72,10 @@ export interface ValidatorOptions {
...
   /**
-   * Fails validation for objects unknown to class-validator. Defaults to false.
+   * Fails validation for objects unknown to class-validator. Defaults to true.
    *
    * For instance, since a plain empty object has no annotations used for validation:
-   * - `validate({})` // passes
+   * - `validate({})` // fails.
    * - `validate({}, { forbidUnknownValues: true })` // fails.
    * - `validate(new SomeAnnotatedEmptyClass(), { forbidUnknownValues: true })` // passes.
    */
```
Comment (braaar) [inline src/validation/ValidatorOptions.ts]: Good catch!

File: test/functional/validator-options.spec.ts [hunk 2]
```diff
@@ -44,4 +44,9 @@ describe('validator options', () => {
       expect(errors.length).toEqual(0);
     });
   });
+
+  it(`should forbidUnknownValues by default`, function () {
+    expect(validator.validateSync({}).length).toEqual(1);
+    expect(validator.validateSync({}, {}).length).toEqual(1);
+  });
```
Comment (braaar) [inline test/functional/validator-options.spec.ts]:
```suggestion
  it(`should returns error on unknown objects if options object argument is passed`, function () {
    const anonymousObject = { badKey: 'This should not pass.' }
    expect(validator.validateSync(anonymousObject, undefined).length).toEqual(1);
    expect(validator.validateSync(anonymousObject, {}).length).toEqual(1);
  });
```

I would suggest that you check something other than just the length. Perhaps do a check for `errors[0].constraints` in addition?

**Review by braaar — CHANGES_REQUESTED** [review]
A few changes, but this looks good!

File: src/validation/ValidatorOptions.ts (hunk 1 above)
Comment (cduff) [reply src/validation/ValidatorOptions.ts]: Thanks. I just added an additional example for further clarity.

File: test/functional/validator-options.spec.ts (hunk 2 above)
Comment (cduff) [reply test/functional/validator-options.spec.ts]: Cool. I just updated that file further based on your feedback and to try to match the style of the rest of the file.

**Comment by cduff** [comment]
Thanks for your review @braaar. I just made changes based on your feedback. Hopefully it's good to merge now?

File: test/functional/validator-options.spec.ts [hunk 3]
```diff
@@ -8,23 +8,20 @@ describe('validator options', () => {
...
-        expect(errors[0].value).toEqual('');
-      });
+    return validator.validate(model, { validationError: { target: false } }).then(errors => {
+      expect(errors.length).toEqual(1);
+      expect(errors[0].target).toBeUndefined();
+      expect(errors[0].property).toEqual('title');
+      expect(errors[0].constraints).toEqual({ isNotEmpty: 'title should not be empty' });
+      expect(errors[0].value).toEqual('');
+    });
   });
```
Comment (braaar) [inline test/functional/validator-options.spec.ts]: These seem like irrelevant changes?

**Review by braaar — COMMENTED** [review]
You should maybe add a test where you pass `{}` as the `options` object

**Comment by cduff** [comment]
> You should maybe add a test where you pass `{}` as the `options` object

@braaar Good idea. Done.

File: test/functional/validator-options.spec.ts (hunk 3 above)
Comment (cduff) [reply test/functional/validator-options.spec.ts]: @braaar The `isActive` and `skipMissingProperties` parts are not relevant to that test so I was just trying to clean it up. I can revert that part if you like?

**Comment by cduff** [comment]
@braaar any other changes needed to get this merged?

File: test/functional/validator-options.spec.ts (hunk 3 above)
Comment (braaar) [reply test/functional/validator-options.spec.ts]: I would submit that as a separate cleanup PR

**Comment by braaar** [comment]
> @braaar any other changes needed to get this merged?

Probably not, but we'll have to wait for @NoNameProvided's review

File: test/functional/validator-options.spec.ts (hunk 3 above)
Comment (cduff) [reply test/functional/validator-options.spec.ts]: Okay @braaar I've reverted the changes not directly related to this PR.

**Comment by cduff** [comment]
@braaar I just made another commit based on your feedback. Let me know if anything else is required to get this merged.

**Comment by cduff** [comment]
Any chance of getting this small PR reviewed & merged @NoNameProvided?

**Comment by cduff** [comment]
@braaar @NoNameProvided Any chance of getting this small PR reviewed & merged?

**Review by braaar — APPROVED** [review]

**Comment by braaar** [comment]
Thanks for the contribution!

**Pull request merged**